client/blocks/login/login-form.jsx

```diff
@@ -105,8 +107,12 @@ export class LoginForm extends Component {
 			! disableAutoFocus && defer( () => this.password && this.password.focus() );
 		}
 
-		if ( requestError.field === 'usernameOrEmail' ) {
+		if ( this.showUsernameError() ) {
 			! disableAutoFocus && defer( () => this.usernameOrEmail && this.usernameOrEmail.focus() );
+		} else {
+			window.location.replace(
```

Comment:
`page.redirect` doesn't trigger a render of the signup component (the `start` controller is not called).